core/templates/dev/head/base-components/base-content.directive.ts

```diff
@@ -40,10 +40,20 @@ angular.module('oppia').directive('baseContent', [
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/base-components/base-content.directive.html'),
       controllerAs: '$ctrl',
-      controller: ['$rootScope', 'BackgroundMaskService',
+      controller: ['$rootScope', '$window', 'BackgroundMaskService',
         'SidebarStatusService', 'UrlService', 'SITE_FEEDBACK_FORM_URL',
-        function($rootScope, BackgroundMaskService,
+        function($rootScope, $window, BackgroundMaskService,
             SidebarStatusService, UrlService, SITE_FEEDBACK_FORM_URL) {
+          // Mimic redirection behaviour in the backend (see issue #7867 for
+          // details).
+          if ($window.location.hostname === 'localhost') {
```

Comment:
@seanlip I think we should be checking for something like: `$window.location.hostname ===oppiaserver.appspot.com`?
From what I see here, we are redirecting from `localhost` to `oppiatestserver`

Reply:
Fixed. Thanks @Jamesjay4199! My mistake.